source/common/upstream/cluster_manager_impl.cc

```diff
@@ -340,24 +343,58 @@ bool ClusterManagerImpl::addOrUpdatePrimaryCluster(const envoy::api::v2::Cluster
   // First we need to see if this new config is new or an update to an existing dynamic cluster.
   // We don't allow updates to statically configured clusters in the main configuration.
   const std::string cluster_name = cluster.name();
-  auto existing_cluster = primary_clusters_.find(cluster_name);
-  if (existing_cluster != primary_clusters_.end() &&
-      (!existing_cluster->second.added_via_api_ ||
-       existing_cluster->second.config_hash_ == MessageUtil::hash(cluster))) {
+  const auto existing_primary_cluster = primary_clusters_.find(cluster_name);
+  const auto existing_warming_cluster = warming_clusters_.find(cluster_name);
+  const uint64_t new_hash = MessageUtil::hash(cluster);
+  if ((existing_primary_cluster != primary_clusters_.end() &&
+       existing_primary_cluster->second->blockUpdate(new_hash)) ||
```

Comment:
There's at least two definitions for "primary cluster" in the ClusterManager world. The first is here: https://github.com/envoyproxy/envoy/blob/e6ff690611b8a3373f6d66066b52d613873e446e/include/envoy/upstream/cluster_manager.h#L33, i.e. the non-TLS cluster definition. The second is here: https://github.com/envoyproxy/envoy/blob/e6ff690611b8a3373f6d66066b52d613873e446e/source/common/upstream/cluster_manager_impl.cc#L196.

Since we're referencing different classes of clusters in this PR, this might be a good opportunity to disambiguate to reduce confusion for the reader.

Reply:
Sure, let me see what I can do.